kotlinsample/src/main/java/com/airbnb/epoxy/kotlinsample/MainController.kt

```diff
@@ -0,0 +1,150 @@
+package com.airbnb.epoxy.kotlinsample
+
+import android.content.Intent
+import android.graphics.Color
+import android.util.Log
+import android.widget.Toast
+import com.airbnb.epoxy.EpoxyController
+import com.airbnb.epoxy.EpoxyModel
+import com.airbnb.epoxy.ModelCollector
+import com.airbnb.epoxy.group
+import com.airbnb.epoxy.kotlinsample.helpers.carouselNoSnapBuilder
+import com.airbnb.epoxy.kotlinsample.models.ItemDataClass
+import com.airbnb.epoxy.kotlinsample.models.ItemViewBindingDataClass
+import com.airbnb.epoxy.kotlinsample.models.carouselItemCustomView
+import com.airbnb.epoxy.kotlinsample.models.coloredSquareView
+import com.airbnb.epoxy.kotlinsample.models.decoratedLinearGroup
+import com.airbnb.epoxy.kotlinsample.models.itemCustomView
+import com.airbnb.epoxy.kotlinsample.models.itemEpoxyHolder
+import com.airbnb.epoxy.kotlinsample.models.itemViewBindingEpoxyHolder
+
+class MainController(
+    private val mainActivity: MainActivity
+) : EpoxyController() {
+
+    override fun buildModels() {
+        group {
+            id("epoxyModelGroupDsl")
+            layout(R.layout.vertical_linear_group)
+
+            coloredSquareView {
+                id("coloredSquareView 1")
+                color(Color.DKGRAY)
+                onVisibilityStateChanged { model, _, visibilityState ->
+                    Log.d(TAG, "$model -> $visibilityState")
+                }
+            }
+
+            coloredSquareView {
+                id("coloredSquareView 2")
+                color(Color.GRAY)
+            }
+
+            coloredSquareView {
+                id("coloredSquareView 3")
+                color(Color.LTGRAY)
+            }
+
+            carouselNoSnapBuilder {
+                id("nested carousel")
+                val lastPage = 10
+                for (i in 0 until lastPage) {
+                    carouselItemCustomView {
+                        id("nested carousel $i")
+                        title("Page $i / $lastPage")
+                        onVisibilityStateChanged { model, _, visibilityState ->
+                            this@MainController.logVisibilityState(model, visibilityState, i)
+                        }
+                    }
+                }
+            }
+        }
+
+        decoratedLinearGroup {
+            id("epoxyModelGroupWithLayoutDsl")
+
+            this@MainController.addColoredSquareView(Color.DKGRAY)
+            this@MainController.addColoredSquareView(Color.GRAY)
+            this@MainController.addColoredSquareView(Color.LTGRAY)
```

Comment:
The problem is here : 

```e: kotlinsample/src/main/java/com/airbnb/epoxy/kotlinsample/MainController.kt: (66, 33): 'fun ModelCollector.addColoredSquareView(color: Int): Unit' can't be called in this context by implicit receiver. Use the explicit one if necessary```

Reply:
It will work with 
```
        decoratedLinearGroup {
            val group = this
            id("epoxyModelGroupWithLayoutDsl")
            with (this@MainController) {
                group.addColoredSquareView(Color.DKGRAY)
                group.addColoredSquareView(Color.GRAY)
                group.addColoredSquareView(Color.LTGRAY)
            }
        }
```
but this is not ideal.